### ISSUE TITLE
Using `applyMiddleware` more than once per store does not work

### ISSUE THREAD

**Issue by mpeyper** [ioof-holdings/redux-subspace#58]
## What happened?

Using redux-subspace's `applyMiddleware` more than once when creating a store causes some of the middleware to not be applied to subspaces.

```javascript
import { createStore } from 'redux'
import { applyMiddleware } from 'redux-subspace'
import thunk from 'redux-thunk'
import wormhole from 'redux-subspace-wormhole'

const store = createStore(reducer, compose(
  applyMiddleware(thunk),
  applyMiddleware(wormhole('example'))
))
```

## What should have happened?

All middleware should be applied to subspaces, regardless of how many calls to `applyMiddleware` are used.

## Any ideas?

Admittedly, it's a pretty uncommon use case to want to use `applyMiddleware` more than once.  

For me, it was trying to create a [react-storybook decorator](https://storybook.js.org/addons/introduction/#1-decorators) that would mount the component within a `SubspaceProvider` and have some default middleware applied while also allowing the additional middleware to be applied.

We managed to work around the issue without needing two calls to `applyMiddleware` but it did highlight this issue to me.

I think it could be overcome by checking to see if the [`store` already has an enhancer in the `subspaceOptions`](https://github.com/ioof-holdings/redux-subspace/blob/master/packages/redux-subspace/src/store/applyMiddleware.js#L14) and composing it with the new enhancer, but that is just a hunch (I haven't looked too closely at it).

**Comment by lkyles1991** [comment]
@mpeyper I was having a look at this one and playing around in applyMiddleware.js (having no luck). Set up a mini test react app but with the same store as above. Interestingly, it started to apply both middlewares when the order was changed i.e
```javascript
const store = createStore(reducer, compose(
  applyMiddleware(wormhole('example')),  
  applyMiddleware(thunk)
))
```

**Comment by mpeyper** [comment]
@lkyles1991 it appears as though all `applyMiddleware` calls are being applied to the root store, but only the first one (because of [how `compose` works](https://github.com/reactjs/redux/blob/master/src/compose.js#L21)) is being applied to subspaces.

Here are some unit tests that highlight the issue:

```js
import { createStore, combineReducers, compose } from 'redux'
import { subspace, applyMiddleware, namespaced } from '../src'

describe('applyMiddleware bug tests', () => {

  const TEST_ACTION = 'TEST_ACTION'

  const testAction = (value) => ({ type: TEST_ACTION, value })

  const childReducer = (state = { value: 'initial value' }, action) => {
    if (action.type === TEST_ACTION) {
      const { type, ...values } = action
      return { ...state, ...values }
    }
    
    return state
  }

  const parentReducer = combineReducers({ child1: childReducer, child2: namespaced('childNamespace')(childReducer) })

  const middleware = (newKey, value) => () => (next) => (action) => {
    if (action.type === TEST_ACTION) {
      return next({ ...action, [newKey]: value })
    }

    return next(action)
  }

  it('should allow multiple applyMiddleware enhances for the root store', () => {
    const parentStore = createStore(parentReducer, compose(
      applyMiddleware(middleware('value2', 'middleware value 1')),
      applyMiddleware(middleware('value3', 'middleware value 2'))
    ))

    parentStore.dispatch(testAction('action value'))

    expect(parentStore.getState()).to.deep.equal({
      child1: {
        value: 'action value',
        value2: 'middleware value 1',
        value3: 'middleware value 2'
      },
      child2: {
        value: 'initial value'
      }
    })
  })

  it('should allow multiple applyMiddleware enhances for subspaces', () => {
    const parentStore = createStore(parentReducer, compose(
      applyMiddleware(middleware('value2', 'middleware value 1')),
      applyMiddleware(middleware('value3', 'middleware value 2'))
    ))

    const childStore = subspace((state) => state.child2, 'childNamespace')(parentStore)

    childStore.dispatch(testAction('action value'))

    expect(parentStore.getState()).to.deep.equal({
      child1: {
        value: 'initial value'
      },
      child2: {
        value: 'action value',
        value2: 'middleware value 1',
        value3: 'middleware value 2'
      }
    })
  })
})
```

Currently, this test results in:

```
  applyMiddleware bug tests
    ✓ should allow multiple applyMiddleware enhances for the root store
    1) should allow multiple applyMiddleware enhances for subspaces


  1 passing (46ms)
  1 failing

  1) applyMiddleware bug tests
       should allow multiple applyMiddleware enhances for subspaces:

      AssertionError: expected { Object (child1, child2) } to deeply equal { Object (child1, child2) }
      + expected - actual

         }
         "child2": {
           "value": "action value"
           "value2": "middleware value 1"
      +    "value3": "middleware value 2"
         }
       }
```

**Comment by mpeyper** [comment]
Resolved by #62
